ast/Trees.h

```diff
@@ -194,15 +194,23 @@ class ExpressionPtr {
     }
 
     explicit operator bool() const noexcept {
-        return get() != nullptr;
+        return ptr != 0;
     }
 
     bool operator==(const ExpressionPtr &other) const noexcept {
-        return get() == other.get();
+        return ptr == other.ptr;
```

Comment:
These pointer comparison operations seem valid to me; I don't think it would be legal for us to have two tagged pointers pointing to the same thing with different tags.